File: destination-city-1436/solution.rb
# LeetCode problem # 1436, 'Destination City'
# https://leetcode.com/problems/destination-city/

# @param {String[][]} paths
# @return {String}
def dest_city(paths)
    (Set.new(paths.map { |p| p[1] }) - paths.map { |p| p[0] }).first
end

File: destination-city-1436/solution.js
/**
 * LeetCode problem # 1436, 'Destination City'
 * https://leetcode.com/problems/destination-city/
 */

/**
 * @param {string[][]} paths
 * @return {string}
 */
function destCity(paths) {
  const set = new Set(paths.flatMap((p) => p));

  for (const p of paths) {
    set.delete(p[0]);
  }

  return [...set][0];
}
